contrib/endpoints/src/api_manager/check_security_rules_test.cc

```diff
@@ -0,0 +1,413 @@
+                               kApis + kAuthentication + kHttp;
+  std::string server_config = kServerConfig;
+  SetUp(service_config, server_config);
+
+  request_context_->set_auth_claims(kJwtEmailPayload);
+  EXPECT_CALL(*raw_env_, DoRunHTTPRequest(AllOf(
+                             Property(&HTTPRequest::url, StrEq(release_url_)),
+                             Property(&HTTPRequest::method, StrCaseEq("GET")))))
+      .WillOnce(Invoke([](HTTPRequest *req) {
+
+        std::map<std::string, std::string> empty;
+        std::string body(kRelease);
+        req->OnComplete(Status::OK, std::move(empty), std::move(body));
+
+      }));
+
+  EXPECT_CALL(*raw_env_,
+              DoRunHTTPRequest(
+                  AllOf(Property(&HTTPRequest::url, StrEq(ruleset_test_url_)),
+                        Property(&HTTPRequest::method, StrCaseEq("POST")))))
+      .WillOnce(Invoke([](HTTPRequest *req) {
+        std::map<std::string, std::string> empty;
+        std::string body;
+        req->OnComplete(Status(Code::INTERNAL, "Cannot talk to server"),
+                        std::move(empty), std::move(body));
+      }));
+
+  CheckSecurityRules(request_context_,
+                     [](Status status) { ASSERT_FALSE(status.ok()); });
+}
+
+TEST_F(CheckSecurityRulesTest, CheckAutzFailWithTestResultFailure) {
+  std::string service_config = std::string(kServiceName) + kProducerProjectId +
+                               kApis + kAuthentication + kHttp;
+  std::string server_config = kServerConfig;
+  SetUp(service_config, server_config);
+
+  request_context_->set_auth_claims(kJwtEmailPayload);
+  EXPECT_CALL(*raw_env_, DoRunHTTPRequest(AllOf(
+                             Property(&HTTPRequest::url, StrEq(release_url_)),
+                             Property(&HTTPRequest::method, StrCaseEq("GET")))))
+      .WillOnce(Invoke([](HTTPRequest *req) {
+
+        std::map<std::string, std::string> empty;
+        std::string body(kRelease);
+        req->OnComplete(Status::OK, std::move(empty), std::move(body));
+
+      }));
+
+  EXPECT_CALL(*raw_env_,
+              DoRunHTTPRequest(
+                  AllOf(Property(&HTTPRequest::url, StrEq(ruleset_test_url_)),
+                        Property(&HTTPRequest::method, StrCaseEq("POST")))))
+      .WillOnce(Invoke([](HTTPRequest *req) {
+        std::map<std::string, std::string> empty;
+        std::string body = kTestResultFailure;
+        req->OnComplete(Status::OK, std::move(empty), std::move(body));
+      }));
+
+  CheckSecurityRules(request_context_, [](Status status) {
+    ASSERT_TRUE(status.CanonicalCode() == Code::PERMISSION_DENIED);
+  });
+}
+
+TEST_F(CheckSecurityRulesTest, CheckAuthzSuccess) {
+  std::string service_config = std::string(kServiceName) + kProducerProjectId +
+                               kApis + kAuthentication + kHttp;
+  std::string server_config = kServerConfig;
+  SetUp(service_config, server_config);
```

Comment:
The SetUp method is part of the test class. But I needed a way to pass configuration to different tests.